internal/surface/backend/ref.go

```
@@ -0,0 +1,768 @@
+// choose.
+func Fingerprint(in IncarnationInput) (ServerID, error) {
+	if in.Endpoint == "" || in.Version == "" || in.Inode == 0 {
+		return ServerID{}, ErrInvalidServerID
+	}
+	h := sha256.New()
+	writeField(h, []byte(in.Endpoint))
+	writeField(h, []byte(in.Version))
+	writeField(h, []byte(in.ServerReported))
+	writeUint(h, in.Device)
+	writeUint(h, in.Inode)
+	writeUint(h, uint64(in.ChangedAtUnixNano))
```

golangci-lint (2.12.2)

[high] G115: integer overflow conversion int64 -> uint64

(gosec)